### ISSUE TITLE
Is Activity model function scopeForSubject correct?

### ISSUE THREAD

**Issue by underdpt** [spatie/laravel-activitylog#89]
Hi, 

Looking for a way to retrieve all ativities for some model, I found the scopeForSubject function:

```php
    /**
     * Scope a query to only include activities for a give subject.
     *
     * @param \Illuminate\Database\Eloquent\Builder $query
     * @param \Illuminate\Database\Eloquent\Model $subject
     *
     * @return \Illuminate\Database\Eloquent\Builder
     */
    public function scopeForSubject(Builder $query, Model $subject): Builder
    {
        return $query->where('subject_id', $subject->getKey());
    }
```

I think that scope doesn't takes into account that I can have the same key for two different models, so it may retrieve some activities that belongs to another model.
I think this function should filter by the model class name (namespace + class) adding something like
```php
->where('subject_type', get_class($subject));
```

**Comment by freekmurze** [comment]
Yup, I'll take care of this when I have the name.

If you need this quick, I'd accept a PR with tests.

**Comment by underdpt** [comment]
I'm sorry I have no experience with tests. 
I added a where clause to my Activity::forSubject() call in the meantime.
Thanks!

**Comment by freekmurze** [comment]
Fixed by @pmatseykanets in `v1.10.1`.
